crates/re_viewer/src/app.rs

```diff
@@ -499,32 +494,70 @@ impl eframe::App for App {
             // Add some margin so that we can later paint an outline around it all.
             main_panel_frame.inner_margin = 1.0.into();
         }
+
+        let this_frame_blueprint_id = self.state.selected_blueprint_id.clone();
+
+        // Materialize the blueprint from the DB if the selected blueprint id isn't
+        // the default one.
+        let blueprint_snapshot = if this_frame_blueprint_id == Default::default() {
+            Default::default()
+        } else {
+            // TODO(jleibs): If the blueprint doesn't exist this probably means we are
+            // initializing a new default-blueprint for the application in question.
+            // Make sure it's marked as a blueprint, and also enable AutoSpaceViews.
+            let blueprint_db = self
+                .log_dbs
+                .entry(this_frame_blueprint_id.clone())
+                .or_insert_with(|| {
+                    let mut blueprint_db = LogDb::default();
+
+                    blueprint_db.add_begin_recording_msg(&BeginRecordingMsg {
+                        row_id: re_log_types::RowId::random(),
+                        info: re_log_types::RecordingInfo {
+                            application_id: this_frame_blueprint_id.as_str().into(),
+                            recording_id: self.state.selected_blueprint_id.clone(),
+                            is_official_example: false,
+                            started: re_log_types::Time::from_seconds_since_epoch(0.0),
```

Comment:
```suggestion
                            started: re_log_types::Time::now(),
```

seems more accurate?

Reply:
This seemed like such a reasonable suggestion until several hours later I was 3 PRs deep fighting with cross-platform time::now support :laughing: 
 - https://github.com/rerun-io/rerun/pull/2090
 - https://github.com/rerun-io/rerun/pull/2093
 - https://github.com/rerun-io/rerun/pull/2094